src/Application/Regulation/Command/DuplicateRegulationCommandHandler.php

```diff
@@ -34,15 +34,28 @@ private function duplicateRegulationOrderRecord(
         Organization $organization,
         RegulationOrder $originalRegulationOrder,
     ): RegulationOrderRecord {
-        $step1Command = new SaveRegulationGeneralInfoCommand();
-        $step1Command->organization = $organization;
-        $step1Command->identifier = $this->translator->trans('regulation.identifier.copy', [
+        $generalInfo = new SaveRegulationGeneralInfoCommand();
+        $generalInfo->organization = $organization;
+        $generalInfo->identifier = $this->translator->trans('regulation.identifier.copy', [
             '%identifier%' => $originalRegulationOrder->getIdentifier(),
         ]);
-        $step1Command->description = $originalRegulationOrder->getDescription();
-        $step1Command->startDate = $originalRegulationOrder->getStartDate();
-        $step1Command->endDate = $originalRegulationOrder->getEndDate();
+        $generalInfo->description = $originalRegulationOrder->getDescription();
+        $generalInfo->startDate = $originalRegulationOrder->getStartDate();
+        $generalInfo->endDate = $originalRegulationOrder->getEndDate();
 
-        return $this->commandBus->handle($step1Command);
+        return $this->commandBus->handle($generalInfo);
+    }
+
+    private function duplicateRegulationLocations(
+        RegulationOrder $originalRegulationOrder,
+        RegulationOrderRecord $duplicatedRegulationOrderRecord,
+    ): void {
+        foreach ($originalRegulationOrder->getLocations() as $location) {
+            $locationCommand = new SaveRegulationLocationCommand($duplicatedRegulationOrderRecord);
+            $locationCommand->address = $location->getAddress();
+            $locationCommand->fromHouseNumber = $location->getFromHouseNumber();
+            $locationCommand->toHouseNumber = $location->getToHouseNumber();
```

Comment:
Sauf erreur, je crois qu'on peut réutiliser le `::create()`

```suggestion
            $locationCommand = SaveRegulationLocationCommand::create($duplicatedRegulationOrderRecord, $location);
```

Reply:
Nop je ne peux pas la réutiliser car la `location` **doit** être null dans le constructeur de la command.